src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3650,7 +3650,7 @@ private void AddSynthesizedRecordMembersIfNecessary(MembersAndInitializersBuilde
             memberNames.Free();
 
             // Synthesizing non-readonly properties in struct would require changing readonly logic for PrintMembers method synthesis
-            Debug.Assert(isRecordClass || !members.Any(m => m is PropertySymbol { GetMethod.IsEffectivelyReadOnly : false }));
+            Debug.Assert(isRecordClass || !members.Any(m => m is PropertySymbol { GetMethod.IsEffectivelyReadOnly: false }));
```

Comment:
This change was made in a separate PR that bumps the CodeStyle analyzers that are used in Roslyn (See https://github.com/dotnet/roslyn/pull/55646/files#diff-d8309181586dfce90074c20deb6e6bc4bf23ae0cf0b9c02b850ee1f9ee055091R3653)

Reply:
Merged from main and rerun with latest